src/Muxlicer.hpp

```diff
@@ -0,0 +1,993 @@
+		if (resetPulseInReceived) {
+			playHeadHasReset = true;
+			runIndex = 8;
+
+			if (playState == STATE_STOPPED) {
+				playState = STATE_PLAY_ONCE;
+			}
+			resetTimer.trigger();
+		}
+
+		// if the play switch has effectively been activated for the first time,
+		// i.e. it's not just still being held
+		const bool switchIsActive = params[PLAY_PARAM].getValue() != STATE_STOPPED;
+		if (playStateTrigger.process(switchIsActive) && switchIsActive) {
+
+			// if we were stopped, check for activation (normal or one-shot)
+			if (playState == STATE_STOPPED) {
+				if (params[PLAY_PARAM].getValue() == STATE_PLAY) {
+					playState = STATE_PLAY;
+				}
+				else if (params[PLAY_PARAM].getValue() == STATE_PLAY_ONCE) {
+					playState = STATE_PLAY_ONCE;
+					runIndex = 8;
+					playHeadHasReset = true;
+				}
+			}
+			// otherwise we are in play mode (and we've not just held onto the play switch),
+			// so check for stop or reset
+			else {
+
+				// top switch will stop
+				if (params[PLAY_PARAM].getValue() == STATE_PLAY) {
+					playState = STATE_STOPPED;
+				}
+				// bottom will reset
+				else if (params[PLAY_PARAM].getValue() == STATE_PLAY_ONCE) {
+					playHeadHasReset = true;
+					runIndex = 8;
+				}
+			}
+		}
+	}
+
+
+	// determines how many gates to yield per step
+	int getGateMode() {
+
+		int gate;
+
+		if (inputs[GATE_MODE_INPUT].isConnected()) {
+			// with gate acting as attenuator, hardware reacts in 1V increments,
+			// where x V -> (x + 1) V yields (x - 1) gates in that time
+			float gateCV = clamp(inputs[GATE_MODE_INPUT].getVoltage(), 0.f, 10.f);
+			float knobAttenuation = rescale(params[GATE_MODE_PARAM].getValue(), -1.f, 8.f, 0.f, 1.f);
+
+			gate = int (floor(gateCV * knobAttenuation)) - 1;
+		}
+		else {
+			gate = (int) roundf(params[GATE_MODE_PARAM].getValue());
+		}
+
+		// should be respected, but make sure
+		gate = clamp(gate, -1, 8);
+
+		if (quadraticGatesOnly) {
+			int quadraticGateIndex = int(floor(rescale(gate, -1.f, 8.f, 0.f, 4.99f)));
+			return possibleQuadraticGates[clamp(quadraticGateIndex, 0, 4)];
+		}
+		else {
+			return gate;
+		}
+	}
+
+	json_t* dataToJson() override {
+		json_t* rootJ = json_object();
+		json_object_set_new(rootJ, "modeCOMIO", json_integer(modeCOMIO));
+		json_object_set_new(rootJ, "quadraticGatesOnly", json_boolean(quadraticGatesOnly));
+		json_object_set_new(rootJ, "allInNormalVoltage", json_integer(allInNormalVoltage));
+		json_object_set_new(rootJ, "mainClockMultDiv", json_integer(mainClockMultDiv.multDiv));
+		json_object_set_new(rootJ, "outputClockMultDiv", json_integer(outputClockMultDiv.multDiv));
+		json_object_set_new(rootJ, "playState", json_integer(playState));
+		json_object_set_new(rootJ, "outputClockFollowsPlayMode", json_boolean(outputClockFollowsPlayMode));
+
+		return rootJ;
+	}
+
+	void dataFromJson(json_t* rootJ) override {
+		json_t* modeJ = json_object_get(rootJ, "modeCOMIO");
+		modeCOMIO = (Muxlicer::ModeCOMIO) json_integer_value(modeJ);
```

Comment:
I'd wrap all these in `if (modeJ)` so the variables are left unchanged rather than being set to 0, false, or 0.0 if there is no `"modeCOMIO"` property.